core/src/main/resources/lib/form/checkbox.jelly

```diff
@@ -87,6 +90,11 @@ THE SOFTWARE.
     <label class="attach-previous ${attrs.title == null ? 'js-checkbox-label-empty' : ''}"
            title="${attrs.tooltip}">${attrs.title}</label>
   </span>
+  <j:if test="${attrs.description != null}">
+    <div class="jenkins-checkbox__description">
+      ${attrs.description}
+    </div>
+  </j:if>
```

Comment:
Please make sure to document changes like this in the proposed changelog. This is a separate feature for developers.